tensorrt_llm/_torch/modules/fused_moe/fused_moe_deepgemm.py

```diff
@@ -358,28 +362,52 @@ def __init__(
         )
 
     def get_workspace(self, m_max: int, group_size: int):
+
+        def get_empty(tensor_shape: list[int], dtype: torch.dtype,
+                      cache_name: str) -> torch.Tensor:
+            if DeepGemmFusedMoE.allocated_buffer_recorder is not None:
+                # Safely get the list of candidates. Defaults to an empty list if key is missing.
+                candidate_buffers = DeepGemmFusedMoE.allocated_buffer_recorder.get(
+                    cache_name, [])
+                numel_like = math.prod(tensor_shape)
+
+                for buffer in candidate_buffers:
+                    numel_buffer = buffer.numel()
+
+                    # buffer just needs to be large enough.
+                    if numel_buffer >= numel_like:
+                        return buffer[0:numel_like].view(
+                            tensor_shape)  # Found a fit, return immediately.
+
+            # If we get here, no suitable buffer was found in the cache. Create a new one.
+            new_buffer = torch.zeros(tensor_shape, device='cuda', dtype=dtype)
+            if DeepGemmFusedMoE.allocated_buffer_recorder is not None:
+                DeepGemmFusedMoE.allocated_buffer_recorder.setdefault(
+                    cache_name, []).append(new_buffer)
```

Comment:
Will this cause more memory consumption since we are allocating and appending new buffers if not fit with existing ones?

Reply:
Thx for the SOL change/fix BTW